---
title: "rota_manuscript_git"
author: "Jesse Rop"
date: "October 30, 2019"
output: 
  html_document:
    theme: default
    highlight: zenburn
    toc: true
    toc_depth: 3
    toc_float: false
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r data_preprocessing, include=FALSE}
library("HardyWeinberg")
library("dplyr")
library("forestplot")
library("SNPassoc")
library("snpStats")
library("sjPlot")
library("sjmisc")
library("sjlabelled")
library("ggplot2")
library("haven")
library("data.table")
library("pander")
library("HardyWeinberg")
library("kableExtra")
library("stargazer")
library("emmeans")
library("magrittr")
library("nnet")
library("lubridate")
library("knitr")
library("broom")
library("arsenal")
library(seqinr)
library(ape)
library(Biostrings)
library(msa)
library("tidyverse")
library("conflicted")
library(ggtree)
library(treeio)
library(phangorn)
library(magrittr)
library(gtsummary)
library(gt)
library("flextable")
library(officer)
library(labelled)

#########
conflict_prefer("filter", "dplyr")
conflict_prefer("%in%", "BiocGenerics")
#conflict_prefer("cbind", "mice")
#########

#############15082019########################READING IN CONTROLS################################################
setwd("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/")
rm(list = ls())

##Declaring variables for use in analysis
a_b_ab_oo = c("A", "B", "AB", "OO")
nonOvsO = c("non_O", "OO")
# rota_strain = c("P8_e", "P4_e", "P6_e")
# rota_strain_q = c(quo(P8_e), quo(P4_e), quo(P6_e))

rota_strain = c("P8_e", "P4_e", "P6_e")
rota_strain_q = c(quo(P8_e), quo(P4_e), quo(P6_e))

rota_years = c("2002", "2003", "2004", "2011", "2012", "2013", "2014")
#rota_years_q = c(quo(2002), quo(2003), quo(2004), quo(2011), quo(2012), quo(2013), quo(2014))
rota_period = c("2002_2004", "2009_July2014")
#rota_period_q = c(quo(`2002_2004`), quo(`2009_July2014`))
snpid = c("secretor_status")
sickle_thal = c( "sickle", "thal")
sickle_thal_q = c( quo(sickle), quo(thal))
hbga_variants = c("rs601338", "rs8176719", "rs8176746", "rs28362459","rs28362458")

###########################################################
#Reading in controls
###########################################################

malariagen_sample <- read.table("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/from local F/malariagen_analysis/Kenya_GWAS_bact_mal.sample", header = T, stringsAsFactors = F)

##Converting variables to appropriate data types
malariagen_sample <- malariagen_sample %>% dplyr::select(-2) %>% mutate(sex = factor(sex, levels = c("M","F","D")),cc_status = factor(cc_status, levels = c("CONTROL", "BACT", "D", "MAL")),mal_sub_status = factor(mal_sub_status, levels = c("CONTROL", "BACT","BOTH","CM", "D","OTHER","SMA")),bact_sub_status = factor(bact_sub_status, levels = c("CONTROL", "ACI","BHS","D","ECOLI","HIB","MAL","NTS","PNEUMO","STAPH")),PLATFORM = factor(PLATFORM)) %>% mutate_if(is.factor,droplevels)

##Reading in list of individuals to be excluded due to failed QC
malariagen_exclusion_list = read.table("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/from local F/malariagen_analysis/MGEN_sex_mismatch.excl", header = F, stringsAsFactors = F)
malariagen_sample_no_dtypes = malariagen_sample %>% filter(!(ID_1 %in% malariagen_exclusion_list$V1))

##reading in bactaeremia sample set with ethnicities for all individuals and thalassaemia genotypes
bact_mal_mgen_final_all_variables = read_csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/malariagen_analysis/bact_mal/bm_mgen_su_gwas_genopheno_final_dset_for_analysis.csv")

##adding ethnicity, rs334 genotypes and Kenyan IDs to the bactaeremia sample set
malariagen_sample_no_dtypes <- bact_mal_mgen_final_all_variables %>% dplyr::select(IID,rs334_all, compressed_ethnicity_all, Kenyan_ID, thall_type) %>% right_join(., malariagen_sample_no_dtypes, by = c("IID" = "ID_1" ))

#reading the plink data into r for analysis
##using read.plink
ABO_FUT2_FUT3_plink_gtypes = read.plink("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/from local F/malariagen_analysis/r_analysis/Kenya_GWAS_bact_mal_b37_ABO_FUT2_FUT3", na.strings = c("00", "-9", "NA") )

gtypes_matrix_minus = mice::cbind(ABO_FUT2_FUT3_plink_gtypes$fam$member, ABO_FUT2_FUT3_plink_gtypes$genotypes@.Data)

##merging genotypes to sample file and subsetting relevant variants from the controls data
colnames(gtypes_matrix_minus)[1] = "IID"

gtypes_plus_pheno = gtypes_matrix_minus %>% data.frame(.) %>% full_join(., malariagen_sample_no_dtypes, by = "IID") %>% mutate_at(vars(6:11), as.numeric)

abo_fut2_fut3_gtypes_plus_pheno = gtypes_plus_pheno %>% 
  dplyr::select(IID, Missing, sex, cc_status, PC1, PC2, PC3, PC4, PC5, PC6, PLATFORM, mal_sub_status, bact_sub_status, rs334_all, compressed_ethnicity_all, Kenyan_ID, thall_type, rs8176746.T, rs8176719.TC, rs601338.A, rs28362459.C, rs28362458.T) %>% 
  mutate_at(vars(c("rs8176746.T", "rs8176719.TC", "rs601338.A", "rs28362459.C", "rs28362458.T")),na_if, "00") %>% 
  mutate(rs601338.A = case_when(rs601338.A == "01" ~ "GG", rs601338.A == "02" ~ "GA", rs601338.A == "03" ~ "AA"),
         rs8176746.T = case_when(rs8176746.T == "01" ~ "GG", rs8176746.T == "02" ~ "GT", rs8176746.T == "03" ~ "TT"),
         rs8176719.TC = case_when(rs8176719.TC == "01" ~ "**", rs8176719.TC == "02" ~ "G*", rs8176719.TC == "03" ~ "GG"),
         rs28362459.C = case_when(rs28362459.C == "01" ~ "AA", rs28362459.C == "02" ~ "AC", rs28362459.C == "03" ~ "CC"),
         rs28362459.C = case_when(rs28362458.T == "01" ~ "CC", rs28362458.T == "02" ~ "CT", rs28362458.T == "03" ~ "CC"))
###########################################################
#END: Reading in controls
###########################################################


#############15082019########################READING IN CASES################################################

##Reading in sickle and thal genotypes and assessing the distribution
sickle_thal_gtypes = read_csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/KCH_RVA_list_2002_2004_ 6_Mar_2018_gn.csv", na = c("NO RESULTS", ""))
sickle_thal_gtypes = sickle_thal_gtypes %>% mutate(sickle = factor(sickle),thal = factor(sickle_thal_gtypes$thal, levels = c("NORM", "HET", "HOMO")))

##Caculating HWE for sickle and thal
sickle_thal_gtypes %>% filter(!is.na(sickle)) %>% dplyr::count(sickle) %>% with(pander(HWChisq(n)))
sickle_thal_gtypes %>% filter(!is.na(thal)) %>% dplyr::count(thal) %>% with(pander(HWChisq(n)))

##Reading in rs601338, rs8176719, rs8176746 and rs28362459 genotypes into a list of tibbles
cases_gtypes = hbga_variants %>% map(~ read_csv(paste("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/genotyping results/",., "_gtypes.csv", sep = ""), na =c("", "Undetermined", "Outlier"))) 

##Merging in rs601338, rs8176719, rs8176746 and rs28362459 genotypes into a 1 table and removing 5 duplicates confirmed to have redundant information
abo_fut2_fut3_rota_gtypes = cases_gtypes %>% purrr::reduce(full_join, by = "serial") %>% mutate(rs601338 = case_when(rs601338 == "GG" ~ "AA", rs601338 == "AA" ~ "GG",rs601338 == "GA" ~ "GA")) %>% dplyr::select(serial, all_of(hbga_variants)) %>% distinct(serial, .keep_all = T)

##Reading in positions and finding positions for repeats renaming names appropriately to allow rowbinding
rota_case_pos_02_04 = read_csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/DNA storage positions/2002-2004 rota positions for all requested samples including those missing in freezer.csv", na =c(""))
rota_case_pos_09_jul14 = read_csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/DNA storage positions/positions for rotavirus case 2009-jul2014 icluding missing.csv", na =c("")) %>% filter(!(duplicated(serialno) & column == "\\N")) %>% dplyr::select(-c(date_brought,spm_id,pk_specimen_id)) %>% rename_all(.,~toupper(.)) %>% dplyr::rename(DATE_SAMPL = SAMPLE_DATE, Biobank_Tray_no = TRAY_LABEL , POS = FK_TRAY, COL = COLUMN, comment =COMMENT) 

rota_case_all_pos = mice::rbind(rota_case_pos_02_04[,-3], rota_case_pos_09_jul14)


rota_cases_gtype_pos = merge(abo_fut2_fut3_rota_gtypes, rota_case_all_pos, by.x = "serial", by.y  = "SERIALNO", all.x = T ) %>% mutate(rs601338_rpts_plus_nsec = case_when(is.na(rs601338) ~ "rpt", rs601338 == "AA" ~ "AA" ),rs8176746_rpts = case_when(is.na(rs8176746) ~ "rpt"),rs8176719_rpts = case_when(is.na(rs8176719) ~ "rpt"),rs28362459_rpts = case_when(is.na(rs28362459) ~ "rpt"),rs28362458_rpts = case_when(is.na(rs28362458) ~ "rpt"))

write.csv(rota_cases_gtype_pos, "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/genotyping repeats/rota_cases_gtype_pos.csv", row.names = F, quote = F)


##incorporating NEW AND CORRECT repeats
abo_fut2_fut3_rota_gtypes_with_all_rpts <- c("rs601338 rpts/rs601338_final_rpts_28112019.csv", "rs8176719 rpts/rota_rs8176719_plt1_rpts_15112019_data.csv", "rs8176719 rpts/rota_rs8176719_rpts_16082019_data.csv",  "rs8176746 rpts/rota_rs8176746_rpts_16082019_data.csv", "pw_459_458/rs28362459_rpts_pw_25032020.csv", "pw_459_458/rs28362458_pw_25032020.csv") %>% map(~read_csv(paste("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/genotyping results/",., sep = ""), na =c("", "Undetermined", "Outlier"))) %>% purrr::reduce(full_join, by = "serial") %>% full_join(rota_cases_gtype_pos, .,  by = "serial") %>% mutate(rs601338 = coalesce(rs601338.y, rs601338.x),rs601338 = case_when(rs601338 == "GA_misgenotyped" | rs601338 == "GA" ~ "GA", TRUE ~ rs601338),rs8176719 = coalesce(rs8176719, rs8176719.x, rs8176719.y), rs8176746 = coalesce(rs8176746.x, rs8176746.y),rs28362458 = coalesce(rs28362458.x, rs28362458.y),rs28362459 = coalesce(rs28362459.x, rs28362459.y)) %>% dplyr::select(serial, hbga_variants)



##23032020end
##Clinical data and strain information for cases

##Reading in 09_14 data from mike (mj), version 2 from nick and version 1 (contributes co-infections) from nick 
rota_cases_info_09_jul14 <- c("rota strains and clinical info 2009_2014_v2.csv", "Rotavirus_2010-2014_mj.csv", "rota strains and clinical info 2009_2014.csv") %>% map(~ read_csv(paste("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/rotavirus infection data/",., sep = ""), na =c("NA", "", ".", "<NA>"))) %>% purrr::reduce(full_join, by = "serial") %>% dplyr::rename("G_Type" = "G_Type.y", "P_Type" = "P_Type.y", "GP" = "Genotype.y") %>% rename_at(.vars = vars(ends_with(".x")), .funs = funs(sub("[.]x$", "", .))) %>%  mutate(P_Type = ifelse(P_Type == "P6", "P[6]", ifelse(P_Type == "P8]", "P[8]", P_Type)),GP = ifelse(GP == "G12P6", "G12P[6]", ifelse(GP == "G12P8]", "G12P[8]", GP))) %>% dplyr::select(c("serial", "admission_date", "weight", "sex", "elisa", "vomiting", "vomiting_durn", "vomiting_episode", "diarrhoea", "diarrhoea_durn", "diarrhoea_episode", "dehydration_status", "shock", "outcome", "pgenotype", "ggenotype", "diarrhoea_24h", "vomitting_24h", "gsequence", "psequence", "muac", "fever", "diagnosis1_disch.y", "diagnosis2_disch.y", "month", "year", "dob", "discharge_date", "loc_name", "sub_loc_name", "ethnic_group", "G_Type", "P_Type", "GP"))%>% rename_at(.vars = vars(ends_with(".y")), .funs = funs(sub("[.]y$", "", .)))

rota_cases_info_02_04 <- c("rota strains and clinical info 2002_2004_v2.csv", "Rotavirus_2002-2004_mj.csv") %>% map(~ read_csv(paste("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/rotavirus infection data/",., sep = ""), na =c("NA", "", ".", "<NA>"))) %>% purrr::reduce(full_join, by = "serial") %>% dplyr::select(-c( "age_months", "date_admission.y", "Year_of_admission", "Epidemic")) %>% rename_at(.vars = vars(ends_with(".x")), .funs = funs(sub("[.]x$", "", .))) %>% rename_at(vars(starts_with("p"), -c(P_Type)), ~toupper(str_replace(.,"([[A-Za-z]])([0-9]+)", "\\1[\\2]"))) %>% rename_at(vars(starts_with("g"), -c(G_Type, GP, gp_type)), ~toupper(.)) %>% mutate_at(vars(gp_type), ~str_replace_all(., c("GNT" = "Gx", "PNT" = "P[x]")))

rota_strains_02_04_mj_duplicated = rota_cases_info_02_04 %>% filter(duplicated(serial))


rota_cases_info_02_04 = rota_cases_info_02_04 %>%  mutate_at( vars(starts_with("G"), -c(G_Type, GP, gp_type)), ~na_if(., "0")) %>%  mutate_at(vars(starts_with("G"),-c(G_Type, GP, gp_type)), funs(ifelse(. == 1, deparse(substitute(.)), NA))) %>% unite(., ggenotype_all_nm, c(starts_with("G"),-c(G_Type, GP, gp_type)), na.rm= T, remove = F, sep = "_") %>% mutate_at(vars(ggenotype_all_nm), ~na_if(.,"")) 

rota_cases_info_02_04 = rota_cases_info_02_04 %>% mutate_at(vars(starts_with("p"), -c(P_Type)), ~na_if(., "0")) %>%  mutate_at(vars(starts_with("p"), -c(P_Type)), funs(ifelse(. == 1, deparse(substitute(.)), NA))) %>% unite(., pgenotype_all_nm, c(starts_with("p"), -c(P_Type)), na.rm= T, remove = F, sep = "_") %>% mutate_at(vars(pgenotype_all_nm), ~na_if(.,"")) %>% dplyr::select(-c("G1", "G2", "G3", "G4", "G8", "G9", "G10", "G3410", "P[4]", "P[6]", "P[8]", "P[9]", "P[11]", "P[14]"))

write_csv(rota_cases_info_02_04, "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_cases_info_02_04.csv")
##uniting the G and P genotypes into a GP genotype variable
rota_cases_info_09_jul14 = unite(rota_cases_info_09_jul14, gp_type, ggenotype, pgenotype, sep = "", remove = F, na.rm = T)


write_csv(rota_cases_info_09_jul14, "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_cases_info_09_jul14.csv")


##selecting variable to bind and binding

rota_cases_info_09_jul14_subset = rota_cases_info_09_jul14 %>% dplyr::select(c("serial", "admission_date", "weight", "sex", "elisa", "vomiting", "vomiting_durn", "vomiting_episode", "diarrhoea", "diarrhoea_durn", "diarrhoea_episode", "shock",  "outcome", "gp_type", "pgenotype", "ggenotype", "gsequence", "psequence", "muac", "fever", "diagnosis1_disch", "diagnosis2_disch", "dob", "discharge_date", "loc_name", "sub_loc_name", "ethnic_group", "G_Type",  "P_Type", "GP") )

# rota_cases_info_09_jul14_subset = rota_cases_info_09_jul14[,c("serial", "admission_date", "weight", "sex", "elisa", "vomiting", "vomiting_durn", "vomiting_episode", "diarrhoea", "diarrhoea_durn", "diarrhoea_episode", "shock",  "outcome", "muac", "fever", "diagnosis1_disch", "diagnosis2_disch", "dob", "discharge_date", "loc_name", "sub_loc_name", "ethnic_group", "gp_type",  "ggenotype", "pgenotype", "ggenotype_all_nm", "pgenotype_all_nm", "gsequence", "psequence", "G_Type",  "P_Type", "GP")]

#rota_cases_info_09_jul14_subset[,c("ggenotype_minus_mixed", "pgenotype_minus_mixed")] = rota_cases_info_09_jul14_subset[,c("ggenotype", "pgenotype")]
#rota_cases_info_09_jul14_subset[,c("rv_status")] = NA
# rota_cases_info_02_04[,c("P[14]")] = NA


##Confirming colnames are matching before binding tables
colnames_df = data.frame("02_04" = colnames(rota_cases_info_02_04[,c("serial", "date_admission", "weight", "sex", "elisa_dako", "vomiting", "vomiting_drn", "episode_vomit", "diarrhoea", "diarrhoea_drn", "episode_diarrhoea", "shock", "disch_type","gp_type", "ggenotype_all_nm", "pgenotype_all_nm", "muac", "fever", "diag1_disch", "diag2_disch", "date_of_birth", "date_disch", "ct_location", "ct_sub_location", "ethnic_group",  "G_Type", "P_Type", "GP")]), "09_14" = c("serial", "admission_date", "weight", "sex", "elisa", "vomiting", "vomiting_durn", "vomiting_episode", "diarrhoea", "diarrhoea_durn", "diarrhoea_episode", "shock",  "outcome", "gp_type", "ggenotype", "pgenotype", "muac", "fever", "diagnosis1_disch", "diagnosis2_disch", "dob", "discharge_date", "loc_name", "sub_loc_name", "ethnic_group", "G_Type",  "P_Type", "GP"))

##Changing the names of the 2002-2004 cases to match those of the 2009-2014 dataset in preparation for binding
rota_cases_info_02_04 = rota_cases_info_02_04 %>% rename_at(vars(c("serial", "date_admission", "weight", "sex", "elisa_dako", "vomiting", "vomiting_drn", "episode_vomit", "diarrhoea", "diarrhoea_drn", "episode_diarrhoea", "shock", "disch_type","gp_type", "ggenotype_all_nm", "pgenotype_all_nm", "muac", "fever", "diag1_disch", "diag2_disch", "date_of_birth", "date_disch", "ct_location", "ct_sub_location", "ethnic_group",  "G_Type", "P_Type", "GP")), ~c("serial", "admission_date", "weight", "sex", "elisa", "vomiting", "vomiting_durn", "vomiting_episode", "diarrhoea", "diarrhoea_durn", "diarrhoea_episode", "shock",  "outcome", "gp_type", "ggenotype", "pgenotype", "muac", "fever", "diagnosis1_disch", "diagnosis2_disch", "dob", "discharge_date", "loc_name", "sub_loc_name", "ethnic_group", "G_Type",  "P_Type", "GP")) %>% dplyr::select(c("serial", "sex", "elisa", "muac", "weight", "shock", "fever", "diarrhoea","diarrhoea_durn", "diarrhoea_episode",  "vomiting","vomiting_durn", "vomiting_episode",  "outcome", "diagnosis1_disch", "diagnosis2_disch","dob","admission_date", "discharge_date",  "loc_name", "sub_loc_name", "ethnic_group", "ggenotype", "pgenotype", "gp_type","G_Type", "P_Type", "GP"))

##formating dates prior to binding since they are of different formats (filtering duplicated individual 64486 as obtained from MJ)
rota_cases_info_02_04 = rota_cases_info_02_04 %>% mutate(dob = parse_date_time(dob, "dmy"),discharge_date = parse_date_time(discharge_date, "mdy"),admission_date = parse_date_time(admission_date, "mdy")) %>% distinct(serial, .keep_all = T)

rota_cases_info_09_jul14_subset =  rota_cases_info_09_jul14_subset %>% mutate(dob = parse_date_time(dob, "mdy"),discharge_date = parse_date_time(discharge_date, "mdy"),admission_date = parse_date_time(admission_date, "mdy"))

##binding the 2 phenotype files and merging ggenotype and pgenotype
rota_cases_all = rota_cases_info_02_04 %>%  bind_rows("2002_2004" = ., "2009_July2014" = rota_cases_info_09_jul14_subset, .id = "surveillance_period" ) %>% unite(., gp_type_all_nm, c(ggenotype, pgenotype), na.rm= T, remove = F, sep = "")  %>% mutate_at(vars(gp_type_all_nm), ~na_if(.,"")) 

##Sorting strains grnotype and writing file for mike an nick
rota_cases_all = rota_cases_all %>% 
  mutate(g_type_nm_exc = case_when(nchar(as.character(ggenotype)) <4 ~ ggenotype),
         p_type_nm_exc = case_when(nchar(as.character(pgenotype)) < 6 ~ pgenotype),
         gtype_all_exc = coalesce(G_Type, g_type_nm_exc),
         ptype_all_exc = coalesce(P_Type, p_type_nm_exc),
         gtype_all_exc = na_if(gtype_all_exc, "Gx"),
         ptype_all_exc = na_if(ptype_all_exc, "P[x]"),
  ) %>% 
  unite(gp_type_nm_exc_mj_all, c(gtype_all_exc, ptype_all_exc), sep = "", remove = F, na.rm = T)  %>%
  mutate(gp_type_nm_exc_mj_all = case_when(nchar(gp_type_nm_exc_mj_all) >4 ~ gp_type_nm_exc_mj_all)) %>%
  mutate_at(vars(ends_with("_exc"),gp_type_nm_exc_mj_all), factor)

rota_geno_pheno = rota_cases_all %>% right_join(.,abo_fut2_fut3_rota_gtypes_with_all_rpts,  by = "serial") %>% left_join(., sickle_thal_gtypes[,c("serial","sickle","thal","age_months")],by = "serial")

##Writing file for mike an nick
rota_strain_gg_discrepancy = rota_geno_pheno %>% mutate(g_type_discrepancies = case_when(ggenotype == "Gx" & ggenotype != G_Type ~ "nm_Gx_mismatch", G_Type == "Gx" & ggenotype != G_Type ~ "mj_Gx_mismatch", nchar(ggenotype) > 3 & ggenotype != G_Type ~ "nm_g_mixed_mismatch",is.na(ggenotype) & nchar(G_Type) < 4 & G_Type != "Gx" ~ "nm_missing", is.na(G_Type) & !is.na(ggenotype) & ggenotype != "Gx" ~ "mj_missing", ggenotype != G_Type ~ "mismatch", ggenotype == G_Type & ggenotype != "Gx" & G_Type != "Gx" ~ "match"),p_type_discrepancies = case_when(pgenotype == "P[x]" & pgenotype != P_Type ~ "nm_P[x]_mismatch", P_Type == "P[x]" & pgenotype != P_Type ~ "mj_P[x]_mismatch", nchar(pgenotype) > 5 & pgenotype != P_Type ~ "nm_p_mixed_mismatch",  is.na(pgenotype) & nchar(P_Type) < 6 & P_Type != "P[x]" ~ "nm_missing", is.na(P_Type) & !is.na(pgenotype) & pgenotype != "P[x]" ~ "mj_missing", pgenotype != P_Type ~ "mismatch",pgenotype == P_Type & pgenotype != "P[x]" & P_Type != "P[x]"  ~ "match"),gp_discrepancies = case_when( is.na(gp_type_all_nm)  ~ "nm_missing", is.na(GP) ~ "mj_missing", gp_type_all_nm != GP ~ "mismatch", gp_type_all_nm == GP ~ "match"),missing_gsequence = case_when( is.na(gsequence)  ~ "missing_gsequence"),missing_psequence = case_when( is.na(psequence)  ~ "missing_psequence")) %>% dplyr::select(serial, surveillance_period,G_Type, ggenotype, g_type_discrepancies,P_Type, pgenotype, p_type_discrepancies,  GP,gp_type_all_nm,    gp_discrepancies, gsequence, missing_gsequence, psequence, missing_psequence) 

rota_strain_gg_discrepancy %>% write_csv(., "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_strain_genotypes.csv")

##030620 Reading in cleaned data from Nickson and Mike
rota_geno_seq_clean = c("rota_strain_genotypes_2002_2004_clean_060320.csv", "rotavirus_genotypes_2009_2014_final_clean_030620.csv") %>% 
  map(~ read_csv(paste("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/rotavirus infection data/",., sep = ""), na =c("NA", "", ".", "<NA>"))) %>%
  purrr::reduce(full_join, by = "serial") %>% 
  mutate(gtype_cln = coalesce(g_typey, ggenotype_cleaned), 
         ptype_cln = coalesce(p_typey, pgenotype_cleaned), 
         g_seq_cln = coalesce(vp7_sequence, gsequence_cleaned), 
         p_seq_cln = coalesce(vp4_sequence, psequence_cleaned)) %>% 
  dplyr::select(serial, gtype_cln, ptype_cln, g_seq_cln, p_seq_cln) %>% 
  unite(., gp_cln, c(gtype_cln, ptype_cln), remove = F, sep = "") %>% 
  mutate_at(vars(g_seq_cln, p_seq_cln), str_to_upper)

##Replacing discrepant with cleaned data
rota_cases_all = rota_cases_all %>% 
  full_join(., rota_geno_seq_clean, by = "serial") %>% 
  mutate(gtype_exc_cln = na_if(gtype_cln, "Gx"),
         ptype_exc_cln = na_if(ptype_cln, "P[x]"),
         gtype_exc_cln = coalesce(gtype_exc_cln, gtype_all_exc),
         ptype_exc_cln = coalesce(ptype_exc_cln, ptype_all_exc)
  ) %>% 
  unite(gp_all_exc, c(gtype_exc_cln, ptype_exc_cln), sep = "", remove = F, na.rm = T) %>%
  mutate_at(vars(ends_with("_exc"),gp_type_nm_exc_mj_all), factor)

##Adding onto clean genotypes mixed and NT strains and creating mixed_nt columns
rota_cases_all = rota_cases_all %>% 
  mutate(ggenotype = coalesce(gtype_cln, ggenotype),
         pgenotype = coalesce(ptype_cln, pgenotype),
         gtype_mx_nt = case_when(nchar(as.character(ggenotype)) >3 | as.character(ggenotype) == "Gx" ~ ggenotype),
         ptype_mx_nt = case_when(nchar(as.character(pgenotype)) >4 | as.character(pgenotype) == "P[x]"~ pgenotype)
  ) %>% 
  unite(GP, c(ggenotype, pgenotype), sep = "", remove = F, na.rm = T) %>%
  mutate_at(vars(ends_with("_mx_nt"), ggenotype, pgenotype, GP), factor)


rota_geno_pheno = rota_cases_all %>% right_join(.,abo_fut2_fut3_rota_gtypes_with_all_rpts,  by = "serial") %>% left_join(., sickle_thal_gtypes[,c("serial","sickle","thal","age_months")],by = "serial")


write_csv(rota_geno_pheno, "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_geno_pheno.csv")

##Writing out missing genotype info
rota_geno_pheno %>% mutate(missing_genotypes = case_when(is.na(gp_cln) ~ "missing_gp", is.na(gtype_cln) ~ "missing_gtype", is.na(ptype_cln) ~ "missing_ptype")) %>% dplyr::select(c("surveillance_period","serial", "gp_cln", "gtype_cln", "ptype_cln", "g_seq_cln","p_seq_cln","missing_genotypes") ) %>% write_csv(., "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_geno_pheno_missing_genotypes.csv")



#############15082019########################MERGING CASES TO CONTROLS $ PREPARING VARIABLES FOR ANALYSIS#####################

##preparing the files for binding cases and control tables
rota_geno_pheno$cc_status = "CASE"


abo_fut2_fut3_gtypes_plus_pheno_controls = abo_fut2_fut3_gtypes_plus_pheno %>% filter(cc_status == "CONTROL") %>% dplyr::select(c("IID", "sex", "cc_status", "rs334_all", "compressed_ethnicity_all", "Kenyan_ID", "thall_type", "rs8176746.T", "rs8176719.TC", "rs601338.A", "rs28362459.C", "rs28362458.T")) %>% dplyr::rename( "sickle"="rs334_all" ,  "ethnic_group"="compressed_ethnicity_all",  "serial" = "Kenyan_ID",  "thal" = "thall_type",  "rs8176746" = "rs8176746.T",  "rs8176719" = "rs8176719.TC",  "rs601338" = "rs601338.A",  "rs28362459" = "rs28362459.C", "rs283624598" = "rs28362458.T")

##Reading in controls dates from gideon
abo_fut2_fut3_gtypes_plus_pheno_controls = read_csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/Gideon extra data for cases and controls/rota_controls_extra_data_gn.csv", na =c("NA", "", ".", "<NA>")) %>% dplyr::select(c("kenyan_id", "date_entry", "dob")) %>% mutate_at(vars("date_entry", "dob"), ~dmy(.)) %>% left_join(abo_fut2_fut3_gtypes_plus_pheno_controls,. , by = c("serial" = "kenyan_id")) %>% dplyr::rename("admission_date" = "date_entry") 

rota_geno_pheno_cc = rota_geno_pheno %>% 
  bind_rows(., abo_fut2_fut3_gtypes_plus_pheno_controls) %>%
  mutate(sex = case_when(sex == "f" | sex == "female" | sex == "F" ~ "female", sex == "m" | sex == "male" | sex == "M" ~ "male"),
         sickle = str_replace_all(sickle, "T", "S"),
         thal = str_to_upper(thal),thal = ifelse(thal == "HOMO", "HOM", thal)) %>%
  mutate_at(vars(c("vomiting", "diarrhoea", "shock", "sex", "ethnic_group")), ~str_to_sentence(.)) %>% 
  mutate(fever = case_when(fever == "y" ~ "Yes", fever == "n" ~ "No"), 
         outcome = case_when(outcome == "Dead" ~ "died",outcome == "Alive" | outcome == "discharge" ~ "discharged", TRUE ~ outcome),
         malaria = ifelse(diagnosis1_disch == "All Types of Malaria" | diagnosis1_disch== "malaria" | diagnosis2_disch == "All Types of Malaria" | diagnosis2_disch == "malaria" , "Malaria", "Others"))



##Generating major strains variable and grouping minor ones
#rota_geno_pheno$strains_comprsd = factor(ifelse(rota_geno_pheno$gp_type == "G1P[8]" | rota_geno_pheno$gp_type == "G9P[8]" | rota_geno_pheno$gp_type == "G8P[4]" | rota_geno_pheno$gp_type == "G2P[4]" | rota_geno_pheno$gp_type == "G8P[6]" , as.character(rota_geno_pheno$gp_type), "OTHER_STRAINS"))
#rota_geno_pheno = merge(rota_geno_pheno, rota_geno_pheno %>% filter(!is.na(strains_comprsd)) %>% mutate(v = 1, strain_p = strains_comprsd) %>% spread(strain_p, v, fill = NA) %>% dplyr::select(c(1,45:50)), by ="serial", all.x = T)
#abo_fut2_fut3_gtypes_plus_pheno_controls[,colnames(rota_geno_pheno)[c(48:54)]] = "CONTROL"

##merging in malaria and bactaeremia phenotypes
rota_geno_pheno_cc = bact_mal_mgen_final_all_variables %>% dplyr::select(c("Kenyan_ID","mal_sub_status", "bact_sub_status" )) %>% left_join(rota_geno_pheno_cc,. , by = c("serial" = "Kenyan_ID"))

rota_geno_pheno_cc = rota_geno_pheno_cc %>% mutate_at(vars(c("rs601338", "rs8176719", "rs8176746", "rs28362459", "sex", "elisa", "vomiting", "diarrhoea", "shock", "outcome", "gp_type", "pgenotype", "ggenotype", "fever", "diagnosis1_disch", "diagnosis2_disch", "loc_name", "sub_loc_name", "ethnic_group", "G_Type", "P_Type", "GP", "sickle", "thal", "cc_status", "malaria", "mal_sub_status", "bact_sub_status")), ~factor(.))


##Translating/Inferring genotypes to user friendly blood group categories/phenotypes
rota_geno_pheno_cc = rota_geno_pheno_cc %>% mutate(phenotypic_bldgrp_6bg = factor(ifelse(rs8176719 == "GG" & rs8176746 == "TT", "BB", ifelse(rs8176719 == "GG" & rs8176746 == "GT", "AB", ifelse(rs8176719 == "GG" & rs8176746 == "GG", "AA", ifelse(rs8176719 == "G*" & rs8176746 == "TT", "BO" , ifelse(rs8176719 == "G*" & rs8176746 == "GT", "BO" , ifelse(rs8176719 == "G*" & rs8176746 == "GG", "AO", ifelse(rs8176719 == "**" , "OO", NA ))))))), levels = c("OO", "AO", "BO", "AA", "BB", "AB")),phenotypic_bldgrp_6bg_4bg = factor(ifelse(rs8176719 == "GG" & rs8176746 == "TT", "B", ifelse(rs8176719 == "GG" & rs8176746 == "GT", "AB", ifelse(rs8176719 == "GG" & rs8176746 == "GG", "A", ifelse(rs8176719 == "G*" & rs8176746 == "TT", "B" , ifelse(rs8176719 == "G*" & rs8176746 == "GT", "B" , ifelse(rs8176719 == "G*" & rs8176746 == "GG", "A", ifelse(rs8176719 == "**" , "OO", NA ))))))), levels = c("OO", "A", "B", "AB")),non_OvsO_jcr = factor(ifelse( phenotypic_bldgrp_6bg == "BB" | phenotypic_bldgrp_6bg == "BO"  | phenotypic_bldgrp_6bg == "AA" | phenotypic_bldgrp_6bg == "AO"  | phenotypic_bldgrp_6bg == "AB", "non_O", "OO"), levels = c("OO", "non_O")))


##2011incorporating ethnicity data from Gideon
gn_extra_cases = read_csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/Gideon extra data for cases and controls/rota_cases_extra_data_gn.csv", na =c("NA", "", ".", "<NA>"))

rota_geno_pheno_cc = gn_extra_cases %>% dplyr::select(c("serial", "ethnic")) %>% left_join( rota_geno_pheno_cc,. , by = "serial") %>% mutate(ethnicity = coalesce(as.character(ethnic_group), str_to_sentence(ethnic)),ethnicity = factor(ifelse(ethnicity == "Chonyi" | ethnicity == "Giriama" | ethnicity == "Kauma", as.character(ethnicity), "Other")))

rota_geno_pheno_cc = rota_geno_pheno_cc[,-c(length(rota_geno_pheno_cc)-1)]


##Creating ABO Secretor combination phenotypes
rota_geno_pheno_cc = rota_geno_pheno_cc %>% mutate(secretor_status = factor(ifelse(rs601338 == "AA", "nonsecretor", "secretor"), levels = c("nonsecretor", "secretor")),OvsnonO_secretor_combi = factor(ifelse(rs601338 != "AA" & non_OvsO_jcr == "OO", "O_secretor", ifelse(rs601338 != "AA" & non_OvsO_jcr == "non_O", "nonO_secretor", ifelse(rs601338 == "AA" & non_OvsO_jcr == "OO", "O_nonsecretor", ifelse(rs601338 == "AA" & non_OvsO_jcr == "non_O", "nonO_nonsecretor", NA)))), levels = c("O_nonsecretor", "nonO_nonsecretor",  "O_secretor", "nonO_secretor")),ABO_secretor_combi = factor(ifelse(rs601338 != "AA" & phenotypic_bldgrp_6bg_4bg == "OO", "O_secretor", ifelse(rs601338 != "AA" & phenotypic_bldgrp_6bg_4bg == "A", "A_secretor", ifelse(rs601338 != "AA" & phenotypic_bldgrp_6bg_4bg == "B", "B_secretor", ifelse(rs601338 != "AA" & phenotypic_bldgrp_6bg_4bg == "AB", "AB_secretor",ifelse(rs601338 == "AA" & phenotypic_bldgrp_6bg_4bg == "OO", "O_nonsecretor", ifelse(rs601338 == "AA" & phenotypic_bldgrp_6bg_4bg == "A", "A_nonsecretor", ifelse(rs601338 == "AA" & phenotypic_bldgrp_6bg_4bg == "B", "B_nonsecretor", ifelse(rs601338 == "AA" & phenotypic_bldgrp_6bg_4bg == "AB", "AB_nonsecretor", NA)))))))), levels = c("O_nonsecretor",   "B_nonsecretor",  "A_nonsecretor",  "AB_nonsecretor",  "O_secretor", "B_secretor", "A_secretor", "AB_secretor")))


##Classifying individuals into surveillance years and periods and calculating age and duration of admission
rota_geno_pheno_cc = rota_geno_pheno_cc %>% mutate(surveillance_year = factor(lubridate::year(rota_geno_pheno_cc$admission_date)),age_years = as.numeric(as.duration(interval(dob, admission_date)), "years"),age_months = as.numeric(as.duration(interval(dob, admission_date)), "months"),hospitalization_days = as.numeric(as.duration(interval(admission_date, discharge_date)), "days"))

#Recording individuals before removing them
cases_for_inc = rota_geno_pheno_cc %>% filter(cc_status == "CASE") %>% dplyr::select(serial) %>% unlist() %>% length()
controls_for_inc = rota_geno_pheno_cc %>% filter(cc_status == "CONTROL") %>% dplyr::select(serial) %>% unlist() %>% length()
cases_under6 = rota_geno_pheno_cc %>% filter(cc_status == "CASE" & !is.na(age_years) & age_years<=6) %>% dplyr::select(serial) %>% unlist() %>% length()
controls_under6 = rota_geno_pheno_cc %>% filter(cc_status == "CONTROL" & !is.na(cc_status)& age_years<=6) %>% dplyr::select(serial) %>% unlist() %>% length()



##ALL THE GENOTYPED CASES ARE IN rota_geno_pheno_cc DATAFRAME UP TO THIS POINT
rota_geno_pheno_cc = rota_geno_pheno_cc %>% filter(age_years<=6 & !is.na(age_years))




##making p4, p6 and p8 inclusive genotypes
rota_geno_pheno_cc = rota_geno_pheno_cc %>% 
  mutate(P4_e = case_when(ptype_exc_cln == "P[4]" ~ "1", cc_status == "CONTROL" ~ "0"),
         P6_e = case_when(ptype_exc_cln == "P[6]" ~ "1", cc_status == "CONTROL" ~ "0"), 
         P8_e = case_when(ptype_exc_cln == "P[8]" ~ "1", cc_status == "CONTROL" ~ "0")
  )
rota_geno_pheno_cc$thal = factor(rota_geno_pheno_cc$thal, levels = c("NORM", "HET", "HOM"))

##preparing final set with genotypes
rota_geno_pheno_cc = rota_geno_pheno_cc %>% filter(!is.na(rs601338) | !is.na(rs8176719) | !is.na(rs8176746) | cc_status == "CONTROL") %>% mutate(
  thal = factor(thal, levels = c("NORM", "HET", "HOM")),
  cc_status = factor(ifelse(cc_status == "CONTROL", "0", "1"), labels = c("Controls", "All rota cases")))

for (i in 1:length(rota_strain)) {
  rota_geno_pheno_cc[,rota_strain[i]] = factor(ifelse(rota_geno_pheno_cc[,rota_strain[i]] == "0", "0", "1"), labels = c("Controls", rota_strain[i]))
}

rota_geno_pheno_cc = rota_geno_pheno_cc %>% mutate(lewis_pos = factor(ifelse(rs28362459 == "AA" & rs28362458 == "TT", "Le+", ifelse(rs28362459 == "AC" & rs28362458 == "TT", "Le+", ifelse(rs28362459 == "AA" & rs28362458 == "TC", "Le+", ifelse(rs28362459 == "CC" & rs28362458 == "TC", "Le-" , ifelse(rs28362459 == "AC" & rs28362458 == "CC", "Le-" ,ifelse(rs28362459 == "CC" | rs28362458 == "CC", "Le-" , ifelse(rs28362459 == "AC" & rs28362458 == "TC", "hets", NA ))))))), levels = c("Le+", "Le-", "hets")),lewis_bg = factor(ifelse(lewis_pos == "Le+" & secretor_status == "secretor", "LeB", ifelse(lewis_pos == "Le+" & secretor_status == "nonsecretor", "LeA", ifelse(lewis_pos == "Le-" & secretor_status == "secretor", "Le-_secretor", ifelse(lewis_pos == "Le-" & secretor_status == "nonsecretor", "Le-_nonsecretor", ifelse(lewis_pos == "hets" & secretor_status == "secretor", "Lehet_secretor", ifelse(lewis_pos == "hets" & secretor_status == "nonsecretor", "Lehet_nonsecretor", NA )))))), levels = c("LeB", "LeA", "Le-_secretor" , "Le-_nonsecretor", "Lehet_secretor", "Lehet_nonsecretor")))%>% set_variable_labels(secretor_status = "Secretor status", phenotypic_bldgrp_6bg_4bg = "ABO 4", phenotypic_bldgrp_6bg = "ABO 6", hospitalization_days = "Hospitalization duration (Days)", weight = "Weight (Kg)", diarrhoea = "Diarrhoea", diarrhoea_durn = "Diarrhoea duration (Days)", vomiting = "Vomiting", vomiting_durn = "Vomiting duration", muac = "Mid upper arm circumference (cm)", ethnicity = "Ethnicity", ptype_exc_cln = "Exclusive RV P genotype", pgenotype = "RV P genotype including mixed", shock = "Shock", fever = "Fever (>37.5C)", malaria = "Malaria co-infection", non_OvsO_jcr = "O VS Non-O", age_years = "Age (years)", sex = "Sex", ptype_mx_nt = "Mixed & non-typable P genotypes")

cases_with_sec = rota_geno_pheno_cc %>% filter(cc_status == "All rota cases" & !is.na(secretor_status)) %>% dplyr::select(serial) %>% unlist() %>% length()
controls_with_sec = rota_geno_pheno_cc %>% filter(cc_status == "Controls" & !is.na(secretor_status)) %>% dplyr::select(serial) %>% unlist() %>% length()

##Removing controls with no secretor or abo status
rota_geno_pheno_cc = rota_geno_pheno_cc %>% filter(!is.na(secretor_status))

cases_with_abo = rota_geno_pheno_cc %>% filter(cc_status == "All rota cases" & !is.na(phenotypic_bldgrp_6bg_4bg)) %>% dplyr::select(serial) %>% unlist() %>% length()
controls_with_abo = rota_geno_pheno_cc %>% filter(cc_status == "Controls" & is.na(phenotypic_bldgrp_6bg_4bg)) %>% dplyr::select(serial) %>% unlist() %>% length()


##Removing controls with no secretor or abo status
# rota_geno_pheno_cc = rota_geno_pheno_cc %>% filter(cc_status == "All rota cases" | (cc_status == "Controls" & !is.na(phenotypic_bldgrp_6bg_4bg)))

#recording those with rota P genotypes
cases_with_pg = rota_geno_pheno_cc %>% filter(cc_status == "All rota cases" & !is.na(pgenotype)) %>% dplyr::select(serial) %>% unlist() %>% length()
cases_with_exc_pg = rota_geno_pheno_cc %>% filter(cc_status == "All rota cases" & !is.na(ptype_exc_cln)) %>% dplyr::select(serial) %>% unlist() %>% length()


##Writing out final dataset
write.csv(rota_geno_pheno_cc, "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_geno_pheno_cc_final.csv", quote = F, row.names = F)
write.csv(rota_geno_pheno_cc %>% filter(cc_status == "All rota cases"), "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/processed data/rota_geno_pheno_cc_cases_final_shipped.csv", quote = F, row.names = F)


```

##Phylogenetic tree for psequences
```{r, echo=F, results= 'asis', warning=F, message=F, eval=F}
p_seq_set = rota_geno_pheno_cc %>% filter(!is.na(p_seq_cln)) %>% dplyr::select( serial, p_seq_cln) %>% tibble::deframe()  %>% DNAStringSet()
p_aln  = msa(p_seq_set)
# msaPrettyPrint(p_aln, y=c(164, 213), output="asis",showNames="none", showLogo="none", askForOverwrite=FALSE)
p_aln_sir <- msaConvert(p_aln, type="seqinr::alignment")
p_dist = dist.alignment(p_aln_sir, "identity")
p_tree = nj(p_dist)
p_ggtree = ggtree(p_tree)

p_tree_annot = rota_geno_pheno_cc %>% filter(!is.na(p_seq_cln)) %>% dplyr::select( serial, secretor_status, OvsnonO_secretor_combi, ABO_secretor_combi, ptype_exc_cln, lewis_pos, lewis_bg, phenotypic_bldgrp_6bg_4bg, non_OvsO_jcr)

p_ggtree_annot = p_ggtree %<+% p_tree_annot 

# p_ggtree_annot + geom_tiplab(aes(linetype=ptype_exc_cln)) + geom_tippoint(aes(size=hospitalization_days, shape=secretor_status), alpha=0.25)
p_ggtree_annot + geom_tippoint(aes(color=ptype_exc_cln), alpha=0.1)
p_ggtree_annot + geom_tippoint(aes(color=ptype_exc_cln, shape = non_OvsO_jcr)) + geom_facet(panel = "dia dur", data = rota_geno_pheno_cc %>% dplyr::select(serial, diarrhoea_durn, hospitalization_days), geom = ggstance::geom_barh, aes(x = diarrhoea_durn),  stat = "identity", width = .6)+ geom_facet(panel = "Trait", data = rota_geno_pheno_cc %>% dplyr::select(serial, diarrhoea_durn, diarrhoea_episode), geom = ggstance::geom_barh, aes(x = diarrhoea_episode),  stat = "identity", width = .6)

tunisia_pendu = read.GenBank(access.nb = paste("MN",836721:836783, sep = ""))
mj_wo = read.GenBank(access.nb = paste("MH40",2782:3560, sep = ""))

##Kilifi and Tunisia sequences


tunisia_pendu_ds = tunisia_pendu %>% as.character %>% lapply(.,paste0,collapse="") %>% unlist
p_tns_klf = c(tunisia_pendu_ds, rota_geno_pheno_cc %>% filter(!is.na(p_seq_cln)) %>% dplyr::select( serial, p_seq_cln) %>% tibble::deframe()) %>% DNAStringSet()
p_tk_aln  = msa(p_tns_klf)
# msaPrettyPrint(p_tk_aln, y=c(164, 213), output="asis",showNames="none", showLogo="none", askForOverwrite=FALSE)
p_tk_aln_sir <- msaConvert(p_tk_aln, type="phangorn::phyDat")
p_tk_dist = dist.ml(p_tk_aln_sir)
p_tk_tree = upgma(p_tk_dist)
p_tk_ggtree = ggtree(p_tk_tree)

p_tk_tree_annot = read.csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/le_pendu_tunisia/serial_numbers_as_accession.csv", na.strings = "?", stringsAsFactors = F) %>% dplyr::select(serial, secretor_status) %>% base::rbind(., p_tree_annot %>% dplyr::select(serial, secretor_status)) %>% mutate(secretor_status = case_when(secretor_status == "Sec" ~ "secretor", secretor_status == "nonsec" ~ "nonsecretor", TRUE ~ secretor_status)) %>% mutate(country = case_when(str_detect(serial,"MN") ~ "Tunisia", TRUE ~ "Kenya")) 

p_tk_ggtree_annot = p_tk_ggtree %<+% p_tk_tree_annot 

# p_tk_ggtree_annot + geom_tiplab(aes(linetype=ptype_exc_cln)) + geom_tippoint(aes(size=hospitalization_days, shape=secretor_status), alpha=0.25)
p_tk_ggtree_annot + geom_tippoint(aes(color=ptype_exc_cln), alpha=0.1)
p_tk_ggtree_annot + geom_tippoint(aes(color=non_OvsO_jcr, shape = ptype_exc_cln))


##ML
p_tk_nj = nj(p_tk_dist)
p_models = modelTest(p_tk_aln_sir, tree = p_tk_nj)
p_env = attr(p_models, "env")
bestmodel <- p_models$Model[which.min(p_models$BIC)]
p_models[order(p_models$BIC),]
p_fitStart = eval(get(bestmodel, p_env), p_env)
#p_fit = optim.pml(p_fitStart, rearrangement = "stochastic",optGamma=TRUE, optInv=TRUE, model="HKY")
p_fit = readRDS("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/p_fit")
#p_bs = bootstrap.pml(p_fit, bs=100, optNni=TRUE)
p_bs = readRDS("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/p_bs")
p_ml_tree = plotBS(midpoint(p_fit$tree), p_bs, p = 50, type="p")

p_ml_tk_ggtree = ggtree(as.treedata(p_ml_tree))

p_ml_tk_ggtree_annot = p_ml_tk_ggtree %<+% p_tk_tree_annot 

# p_tk_ggtree_annot + geom_tiplab(aes(linetype=ptype_exc_cln)) + geom_tippoint(aes(size=hospitalization_days, shape=secretor_status), alpha=0.25)
# p_tk_ggtree_annot + geom_tippoint(aes(color=ptype_exc_cln), alpha=0.1)
# p_tk_ggtree_annot + geom_tippoint(aes(color=non_OvsO_jcr, shape = ptype_exc_cln))

p_ml_tk_ggtree_annot + geom_tippoint(aes(color=country, shape = secretor_status, size = ptype_exc_cln))
p_ml_tk_ggtree_annot + geom_tippoint(aes(color=country, shape = secretor_status)) + geom_nodelab()

```

```{r, echo=F, results= 'asis', warning=F, message=F}

p_tree_annot = rota_geno_pheno_cc %>% filter(!is.na(p_seq_cln)) %>% dplyr::select( serial, secretor_status, OvsnonO_secretor_combi, ABO_secretor_combi, ptype_exc_cln, lewis_pos, lewis_bg, phenotypic_bldgrp_6bg_4bg, non_OvsO_jcr)

p_tk_tree_annot = read.csv("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/Rotavirus cases/le_pendu_tunisia/serial_numbers_as_accession.csv", na.strings = "?", stringsAsFactors = F) %>% 
  dplyr::select(serial, secretor_status) %>% 
  dplyr::bind_rows(., p_tree_annot %>% 
                     dplyr::select(serial, secretor_status, ptype_exc_cln) %>% 
                     mutate_all(as.character)) %>% 
  mutate(secretor_status = case_when(secretor_status == "Sec" ~ "secretor", secretor_status == "nonsec" ~ "nonsecretor", TRUE ~ secretor_status),
         Country = case_when(str_detect(serial,"MN") ~ "Tunisia", TRUE ~ "Kenya")) %>%
  mutate_at(vars(secretor_status), str_to_sentence)  %>%
  tidyr::replace_na(list(`secretor_status` = "Missing", `ptype_exc_cln` = "Missing"))#%>%
  # dplyr::rename(c(`Secretor status` = "secretor_status", `P genotype` = "ptype_exc_cln"))

p_fit = readRDS("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/p_fit")
p_bs = readRDS("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/p_bs")
p_ml_tree = plotBS(midpoint(p_fit$tree), p_bs, p = 50, type="p")

p_ml_tk_ggtree = ggtree(as.treedata(p_ml_tree))

p_ml_tk_ggtree_annot = p_ml_tk_ggtree %<+% p_tk_tree_annot 

p_ml_plot = p_ml_tk_ggtree_annot + geom_tippoint(aes(color=ptype_exc_cln, shape = secretor_status)) + geom_cladelabel(node=381, label="Tunisia") + geom_cladelabel(node=308, label="Tunisia") + geom_cladelabel(node=342, label="Tunisia") +  scale_shape_manual(name = "Secretor status", values=c(2,8,1), breaks = c("Missing", "Nonsecretor", "Secretor"), labels = c("Missing", "Non-secretor", "Secretor")) + scale_color_manual(name = "P genotype", values = c("red", "blue","green","grey"), breaks = c("Missing", "P[4]", "P[6]", "P[8]"), labels = c("Missing", "P[4]", "P[6]", "P[8]")) + geom_cladelabel(node=381, label="Tunisia") + geom_cladelabel(node=308, label="Tunisia") + geom_cladelabel(node=342, label="Tunisia")

read_docx("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx") %>%  
  body_add_par(value = "Table 1: FUT2 sequencing participants", style = "heading 2") %>% 
  body_add_gg(p_ml_plot, width = 6, height = 8, res = 500 ) %>%  
  body_add_par(value = "Maximum likelihood tree of RVGA P sequences from Kilifi and Tunisian population obtained from Secretors and Non-secretors. Tunisian strains are labelled with the vertical bar and the unlabelled ones are from Kenyan", style = "Table Caption") %>% 
  body_end_section_continuous() %>% 
  print("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx")

```


```{r, echo=F, results= 'asis', warning=F, message=F, eval=F}
##Effect of sickle and thal before removal of sicklers {.tabset}
panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)
rota_vs_sickle_thal = function(){
  for (i in 1:length(rota_strain)){
    cat('###', rota_strain[i], "\n")
    model_list=list()
    count_list=list()
    for (k in 1:length(sickle_thal)){
      ##WITH MALARIA: rota~ sickle/thal + ethnicity
      model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  sickle_thal[k], "+" ,  paste("ethnicity", collapse = "+") )),data = rota_geno_pheno_cc , family = "binomial")
      count_list[[length(count_list)+1]] = rota_geno_pheno_cc %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[k]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[k]]) %>% mutate(id=row_number())
      
      ##MINUS MALARIA: rota~ sickle/thal + ethnicity
      model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  sickle_thal[k], "+" ,  paste("ethnicity", collapse = "+") )),data = rota_geno_pheno_cc[rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"]),] , family = "binomial")
      count_list[[length(count_list)+1]] = rota_geno_pheno_cc[rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"]),] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[k]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[k]]) %>% mutate(id=row_number())
      
    }
    
    ##WITH MALARIA: rota~ sickle+thal + ethnicity
    model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  paste(sickle_thal, collapse = "+"), "+" ,  paste("ethnicity", collapse = "+") )),data = rota_geno_pheno_cc , family = "binomial")
    count_list[[length(count_list)+1]] = rota_geno_pheno_cc %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[1]]) & !is.na(!!sickle_thal_q[[2]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[1]], !!sickle_thal_q[[2]]) %>% mutate(id=row_number())
    
    ##WITH MALARIA: rota~ sickle+thal + ethnicity
    model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  paste(sickle_thal, collapse = "+"), "+" ,  paste("ethnicity", collapse = "+") )),data = rota_geno_pheno_cc[rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"]),] , family = "binomial")
    count_list[[length(count_list)+1]] = rota_geno_pheno_cc[rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"]),]  %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[1]]) & !is.na(!!sickle_thal_q[[2]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[1]], !!sickle_thal_q[[2]]) %>% mutate(id=row_number())
    
    
    cat("\n")    
    
    cat(paste("<B><U> ", rota_strain[i], " ~ ",  paste(sickle_thal, collapse = " (+ or /) "), " + " , paste("ethnicity", collapse = "+"), "</U></B> \n \n"))
    tab_model(model_list, dv.labels = paste(rep(c("Plus mal+ve |", "Minus mal+ve |"), times = 3),c(rep(sickle_thal, times = 1, each = 2),rep("sickle & thal",2))), use.viewer = F, rm.terms = c("Chonyi", "Giriama", "Kauma", "Other"), collapse.ci = TRUE, show.reflvl = TRUE) %>% return() %$% knitr %>% pander()
    cat("\n")  
    #pander(bind_cols(count_list))
    ##WITH MALARIA: rota~ sickle+thal + ethnicity
    
    #cat(paste("<B><U> ", rota_strain[i], "~",  paste(sickle_thal, collapse = "+/"), "+" , paste("ethnicity", collapse = "+"), "</U></B> \n \n"))
    cat("<B><U> Counts of all rotavirus-infected individuals in the model </U></B> \n \n")
    
    pander(setNames(subset(Reduce(function(x, y, ...) merge(x, y, all = TRUE, by = "id", ...),  count_list), select=-id), c(paste(rep(c("Plus mal+ve |", "Minus mal+ve |"), times = 2, each = 3), c(rep(sickle_thal, times = 1, each = 6))), paste(rep(c("Plus mal+ve |", "Minus mal+ve |"), times = 1, each = 4), c(rep("sickle & thal",times = 1, each = 8)))))) 
    cat("\n")
  }
}
rota_vs_sickle_thal()
```


```{r echo=F, results= 'asis', warning=F, message=F, eval=F}
##Effect of sickle and thal before removal of sicklers split by surveillance period {.tabset}

panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)
panderOptions('evals.messages', FALSE)

rota_vs_sickle_thal_accounting_time = function(surveillance, year_or_period){
  
  for (i in 1:length(rota_strain)){
    cat('###', rota_strain[i], "\n")
    for (k in 1:length(sickle_thal)){
      #cat(sickle_thal[k], "\n \n")
      model_list=list()
      count_list=list()
      mal_list=list()
      
      for (j in 1:length(year_or_period)){
        ##rota strain ~ sickle/thal + ethnicity in all plus those with malaria
        model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  sickle_thal[k], "+" ,  paste("ethnicity", collapse = "+") )),data = rota_geno_pheno_cc[rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls",] , family = "binomial")
        ##counting those all including those with malaria
        count_list[[length(count_list)+1]] = rota_geno_pheno_cc[rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls",] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[k]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[k]]) %>% mutate(id=row_number())
        
        ##rota strain ~ sickle/thal + ethnicity in those without malaria
        model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  sickle_thal[k], "+" ,  paste("ethnicity", collapse = "+") )),data = rota_geno_pheno_cc[(rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"])) & (rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls"),] , family = "binomial")
        ##counting those without malaria
        count_list[[length(count_list)+1]] = rota_geno_pheno_cc[(rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"])) & (rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls"),] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[k]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[k]]) %>% mutate(id=row_number())
        ##counting those with malaria
        mal_list[[length(mal_list)+1]] = rota_geno_pheno_cc[rota_geno_pheno_cc[,"malaria"] =="Malaria" & rota_geno_pheno_cc[, surveillance] == year_or_period[j],] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[k]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[k]]) %>% mutate(id=row_number())
        
      }
      
      cat("\n")    
      cat(paste("<B><U> ", rota_strain[i], " ~ ",  paste(sickle_thal[k], collapse = " + "), " + " , paste("ethnicity", collapse = " + "), "</U></B> \n \n"))
      
      tab_model(model_list, dv.labels = paste(rep(c("Plus mal+ve |", "Minus mal+ve |"), times = length(year_or_period)),rep(year_or_period, times = 1, each = 2)), use.viewer = F, rm.terms = c("Chonyi", "Giriama", "Kauma", "Other"), collapse.ci = TRUE, show.reflvl = TRUE)  %>% return() %$% knitr %>% pander()
      cat("\n \n \n \n")
      #pander(full_join(unlist(count_list), by = "id"))
      
      cat("<B><U> Counts of all rotavirus-infected individuals in the model </U></B> \n \n")
      pander(setNames(subset(Reduce(function(x, y, ...) merge(x, y, all = TRUE, by = "id", ...),  count_list), select=-id), c(paste(rep(c("Including malaria -", "Minus malaria -"), times = length(year_or_period), each = 3),rep(year_or_period, times = 1, each = 6)))))
      cat("\n \n \n \n")
      
      cat("<B><U> Counts of individuals co-infected with rotavirus & malaria in the model </U></B>  \n \n")
      pander(setNames(subset(Reduce(function(x, y, ...) merge(x, y, all = TRUE, by = "id", ...),  mal_list), select=-id), c(rep(year_or_period, times = 1, each = 3))))
      cat("\n \n \n \n")
      
    }
    
    model_list=list()
    count_list=list()
    mal_list=list()
    
    for (j in 1:length(year_or_period)){
      
      ##rota strain ~ sickle + thal + ethnicity in those without malaria
      model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  paste(sickle_thal, collapse = "+"), "+" ,  paste("ethnicity", collapse = "+") )), data = rota_geno_pheno_cc[rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls",] , family = "binomial")
      count_list[[length(count_list)+1]] = rota_geno_pheno_cc[rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls",] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[1]]) & !is.na(!!sickle_thal_q[[2]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[1]], !!sickle_thal_q[[2]]) %>% mutate(id=row_number())
      
      ##rota strain ~ sickle + thal + ethnicity in those without malaria
      model_list[[length(model_list)+1]] = glm(as.formula(paste(rota_strain[i], "~",  paste(sickle_thal, collapse = "+"), "+" ,  paste("ethnicity", collapse = "+") )), data = rota_geno_pheno_cc[(rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"])) & (rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls"),] , family = "binomial")
      count_list[[length(count_list)+1]] = rota_geno_pheno_cc[(rota_geno_pheno_cc[,"malaria"] !="Malaria"| is.na(rota_geno_pheno_cc[,"malaria"])) & (rota_geno_pheno_cc[, surveillance] == year_or_period[j] | rota_geno_pheno_cc[, "cc_status"] == "Controls"),] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[1]]) & !is.na(!!sickle_thal_q[[2]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[1]], !!sickle_thal_q[[2]]) %>% mutate(id=row_number())
      
      ##counting those with malaria
      mal_list[[length(mal_list)+1]] = rota_geno_pheno_cc[rota_geno_pheno_cc[,"malaria"] =="Malaria" & rota_geno_pheno_cc[, surveillance] == year_or_period[j],] %>% filter(!is.na(ethnicity) & !is.na(!!rota_strain_q[[i]]) & !is.na(!!sickle_thal_q[[1]]) & !is.na(!!sickle_thal_q[[2]]))  %>% dplyr::count(!!rota_strain_q[[i]], !!sickle_thal_q[[1]], !!sickle_thal_q[[2]]) %>% mutate(id=row_number())
      
    }
    
    cat("\n")    
    cat(paste("<B><U> ", rota_strain[i], " ~ ",  paste(sickle_thal[k], collapse = " + "), " + " , paste("ethnicity", collapse = " + "), "</U></B> \n \n"))
    
    tab_model(model_list, dv.labels = paste(rep(c("Plus mal+ve | sickle & thal |", "Minus mal+ve | sickle & thal |"), times = length(year_or_period)),rep(year_or_period, times = 1, each = 2)), use.viewer = F, rm.terms = c("Chonyi", "Giriama", "Kauma", "Other"), collapse.ci = TRUE, show.reflvl = TRUE)  %>% return() %$% knitr %>% pander()
    cat("\n \n \n \n")
    
    cat("<B><U> Sickle & thal counts of all cases and controls in the model </U></B> \n \n")
    pander(setNames(subset(Reduce(function(x, y, ...) merge(x, y, all = TRUE, by = "id", ...),  count_list), select=-id), c(paste(rep(c("Including malaria -", "Minus malaria -"), times = length(year_or_period), each = 4),rep(year_or_period, times = 1, each = 8)))))
    cat("\n \n \n \n")
    
    cat("<B><U> Sickle & thal counts of cases co-infected with rotavirus & malaria in the model </U></B>  \n \n")
    pander(setNames(subset(Reduce(function(x, y, ...) merge(x, y, all = TRUE, by = "id", ...),  mal_list), select=-id), c(rep(year_or_period, times = 1, each = 3))))
    cat("\n \n \n \n")
    
    cat("\n \n")
  }
}
rota_vs_sickle_thal_accounting_time(surveillance = "surveillance_period", year_or_period = rota_period[1])
```


```{r, echo=F, results= 'asis', warning=F, message=F, eval=F}
##Effect of sickle and thal before removal of sicklers split by surveillance year {.tabset}

panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

rota_vs_sickle_thal_accounting_time(surveillance = "surveillance_year", year_or_period = rota_years[1:3])
```


```{r further data_preprocessing and tabulation, echo=F, results= 'hide', warning=F}
##HWE and Chisqr tests to ensure data validity

panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

##removing sicklers before analysis
rota_geno_pheno_cc = rota_geno_pheno_cc[rota_geno_pheno_cc[,"sickle"] != "SS" | is.na(rota_geno_pheno_cc[,"sickle"]), ]
rota_geno_pheno_cc$sickle = droplevels(rota_geno_pheno_cc$sickle)
rota_geno_pheno_cc$thal = factor(rota_geno_pheno_cc$thal, levels = c("NORM", "HET", "HOM"))

#############08102019########################ASSOCIATION ANALYSIS REPLICATING MALGEN################################################

##HWE
##rs601338
##distribution of secretors in cases and controls
pander(addmargins(table(rota_geno_pheno_cc$cc_status, rota_geno_pheno_cc$secretor_status)))
pander(prop.table(table(rota_geno_pheno_cc$cc_status, rota_geno_pheno_cc$secretor_status), 1))
rota_geno_pheno_cc %>% filter(!is.na(rs601338)) %>% dplyr::count(rs601338)
rota_geno_pheno_cc %>% filter(!is.na(rs601338)) %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs601338) & cc_status == "Controls") %>% dplyr::count(rs601338)
rota_geno_pheno_cc %>% filter(!is.na(rs601338) & cc_status == "Controls") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs601338) & cc_status == "All rota cases") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))

##HWE calculation in the 3 hbga SNPS in controls with and those without ethnicity
rota_geno_pheno_cc %>% filter(!is.na(rs601338) & cc_status == "Controls") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs601338) & cc_status == "Controls" & !is.na(ethnicity)) %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs8176719) & cc_status == "Controls" ) %>% dplyr::count(rs8176719) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs8176719) & cc_status == "Controls" & !is.na(ethnicity)) %>% dplyr::count(rs8176719) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs8176746) & cc_status == "Controls") %>% dplyr::count(rs8176746) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter(!is.na(rs8176746) & cc_status == "Controls" & !is.na(ethnicity)) %>% dplyr::count(rs8176746) %>% with(pander(HWChisq(n)))

pander(addmargins(table(rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "All rota cases", "ethnicity"])))
pander(prop.table(table(rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "All rota cases", "ethnicity"])))
pander(addmargins(table(rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "Controls", "ethnicity"])))
pander(prop.table(table(rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "Controls", "ethnicity"])))
pander(addmargins(table(rota_geno_pheno_cc[, "ethnicity"])))
pander(prop.table(table(rota_geno_pheno_cc[, "ethnicity"])))


##distribution of secretor status in all individuals and in controls by ethnicity and calculation of secretor HWE within each 
pander(addmargins(table(rota_geno_pheno_cc[, c("ethnicity", "secretor_status")])))
pander(prop.table(table(rota_geno_pheno_cc[, c("ethnicity", "secretor_status")]), margin = 1))

###in controls
pander(addmargins(table(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "Controls", c("ethnicity", "secretor_status")])))
pander(prop.table(table(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "Controls", c("ethnicity", "secretor_status")]), margin = 1))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "Controls" & ethnicity == "Chonyi") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "Controls" & ethnicity == "Giriama") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "Controls" & ethnicity == "Kauma") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "Controls" & ethnicity == "Other") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
###in cases
pander(addmargins(table(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "All rota cases", c("ethnicity", "secretor_status")])))
pander(prop.table(table(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "All rota cases", c("ethnicity", "secretor_status")]), margin = 1))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "All rota cases" & ethnicity == "Chonyi") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "All rota cases" & ethnicity == "Giriama") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "All rota cases" & ethnicity == "Kauma") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))
rota_geno_pheno_cc %>% filter( !is.na(rs601338) & cc_status == "All rota cases" & ethnicity == "Other") %>% dplyr::count(rs601338) %>% with(pander(HWChisq(n)))

# pander(tidy(chisq.test(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "Controls" & !is.na(rota_geno_pheno_cc[,"cc_status"]), "secretor_status"], rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "Controls" & !is.na(rota_geno_pheno_cc[,"cc_status"]), "ethnicity"])))
# pander(tidy(chisq.test(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "All rota cases", "secretor_status"], rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "All rota cases", "ethnicity"])))
# 
# pander(tidy(chisq.test(rota_geno_pheno_cc[, "cc_status"], rota_geno_pheno_cc[, "ethnicity"])))
# pander(tidy(chisq.test(rota_geno_pheno_cc[, "secretor_status"], rota_geno_pheno_cc[, "ethnicity"])))
# 
# #t.test(rota_geno_pheno_cc[, "age_months"] ~ rota_geno_pheno_cc[, "cc_status"])
# #oneway.test(rota_geno_pheno_cc[, "age_months"] ~ rota_geno_pheno_cc[, "secretor_status"])
# 
# pander(prop.table(table(rota_geno_pheno_cc[!is.na(rota_geno_pheno_cc[, "secretor_status"]), c("secretor_status", "phenotypic_bldgrp_6bg_4bg")]), margin = 1))
# pander(tidy(chisq.test(rota_geno_pheno_cc[, "secretor_status"], rota_geno_pheno_cc[, "phenotypic_bldgrp_6bg_4bg"])))
# pander(tidy(chisq.test(rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "All rota cases", "secretor_status"], rota_geno_pheno_cc[rota_geno_pheno_cc[, "cc_status"] == "All rota cases", "phenotypic_bldgrp_6bg_4bg"])))
# 
# ##13052019 Number of cases and controls in individuals included in logistic regression model i.e with complete case, sex, ethnicity, hbb_rs334 and secretor status info
# 

#bgs = c("rs601338", "rs8176719",  "rs8176746", "phenotypic_bldgrp_6bg_4bg", "rs601338 * phenotypic_bldgrp_6bg_4bg", "thal", "sickle")

for (i in 1:length(rota_strain)) {
  print(rota_strain[i])
  pander(addmargins(table(rota_geno_pheno_cc[!is.na(rota_geno_pheno_cc[,"sickle"]) & !is.na(rota_geno_pheno_cc[, "phenotypic_bldgrp_6bg_4bg"]),rota_strain[i]])))
}

for (i in 1:length(rota_strain)) {
  print(rota_strain[i])
  pander(addmargins(table(rota_geno_pheno_cc[ !is.na(rota_geno_pheno_cc[, "sickle"]) & !is.na(rota_geno_pheno_cc[, "secretor_status"]) ,rota_strain[i]])))
}

for (i in 1:length(rota_strain)) {
  print(rota_strain[i])
  pander(addmargins(table(rota_geno_pheno_cc[ !is.na(rota_geno_pheno_cc[, "sickle"]) & !is.na(rota_geno_pheno_cc[, "ABO_secretor_combi"]) ,rota_strain[i]])))
}
```


```{r, echo=F, results= 'asis', warning=F, message=F, eval=F}
##Effect of sickle and thal after removal of sicklers {.tabset}

panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

rota_vs_sickle_thal()
```


```{r echo=F, results= 'asis', warning=F, message=F, eval=F}
##Effect of sickle and thal after removal of sicklers split by surveillance period {.tabset}

panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

rota_vs_sickle_thal_accounting_time(surveillance = "surveillance_period", year_or_period = rota_period[1])
```


```{r, echo=F, results= 'asis', warning=F, message=F, eval=F}
##Effect of sickle and thal after removal of sicklers split by surveillance year {.tabset}

panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

rota_vs_sickle_thal_accounting_time(surveillance = "surveillance_year", year_or_period = rota_years[1:3])
```

##Table 1: Comparing 2 surveillance periods {.tabset}
```{r, echo=F, results= 'asis', warning=F, message=F}
panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

## Flowchart of participants
  inclussion_flowchart = DiagrammeR::grViz("
  digraph graph2 {
  
  graph [layout = dot]
  
  # node definitions with substituted label text
  node [shape = rectangle, width = 4, fillcolor = Biege]
  a [label = '@@1']
  b [label = '@@2']
  c [label = '@@3']
  d [label = '@@4']
  e [label = '@@5']
  
  a -> b -> c -> d -> e
  
  }
  
  [1]:  paste0('Participants selected for inclusion (Cases = ', cases_for_inc, ') (Controls = ', controls_for_inc, ')')
  [2]: paste0('Individuals under 6 years of age (Cases = ', cases_under6, ') (Controls = ', controls_under6, ')')
  [3]: paste0('Individuals with secretor status (Cases = ', cases_with_sec, ') (Controls = ', controls_with_sec, ')')
  [3]: paste0('Individuals with abo status (Cases = ', cases_with_abo, ') (Controls = ', controls_with_abo, ')')
  [4]: paste0('Individuals with P genotypes (Cases = ', cases_with_pg, ')')
  [5]: paste0('Individuals with exclusive P genotypes and mixed P genotypes (Cases = ', cases_with_exc_pg, ')')
  ")
  
html_print(inclussion_flowchart) %>% webshot(file = 'C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/inclusion_flow.png', delay = 10,selector = '.html-widget-static-bound')

read_docx("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx") %>%  
  body_add_par(value = "Figure 1: Case control participants inclusion flow-chart", style = "heading 2") %>% 
  body_add_img(src = "C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/inclusion_flow.png", width = 6, height = 3) %>%  
  body_add_par(value = "Details on elimination of case control into the study", style = "Image Caption") %>% 
  body_end_section_continuous() %>% print("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx")


##Preparing table 1 split by surveillance period
# summary(tableby(surveillance_period ~ age_years + hospitalization_days + weight + diarrhoea + vomiting + muac + ethnicity + ptype_exc_cln + secretor_status + non_OvsO_jcr + phenotypic_bldgrp_6bg_4bg + pgenotype + lewis_bg + lewis_pos,data = rota_geno_pheno_cc, numeric.stats = c("Nmiss", "median", "iqr", "range", "medianmad", "medianq1q3"), cat.stats=c("Nmiss", "countrowpct")), pfootnote = T, digits = 2, digits.pct =2, conf.level = 0.95)
# 
summary(tableby(surveillance_year ~ ptype_exc_cln + pgenotype,data = rota_geno_pheno_cc, numeric.stats = c("Nmiss", "median", "iqr", "range", "medianmad", "medianq1q3"), cat.stats=c("Nmiss", "countrowpct")), pfootnote = T, digits = 2, digits.pct =2, conf.level = 0.95)

summary(tableby(secretor_status ~ hospitalization_days + weight + diarrhoea + diarrhoea_durn + vomiting + vomiting_durn + muac + ethnicity + ptype_exc_cln + non_OvsO_jcr + phenotypic_bldgrp_6bg_4bg + pgenotype + shock + fever + malaria, data = rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "All rota cases",], numeric.stats = c("Nmiss",  "medianq1q3"), cat.stats=c("Nmiss", "countrowpct")), pfootnote = T, digits = 2, digits.pct =2, conf.level = 0.95)

# results_tbl = read_docx()

# this function adds footnotes to each pvalue of the test performed
add_p_footnotes <- function(x) {
  # grouping data, one line per type of test performed
  prep_to_create_code <-
    x$meta_data %>%
    dplyr::filter(!is.na(stat_test_lbl) & stat_test_lbl != "") %>%
    dplyr::select(var_label, stat_test_lbl) %>%
    dplyr::group_nest(stat_test_lbl)
  
  # other calls to apply to x
  prepping_calls <-
    list(rlang::expr(modify_footnote(x, p.value ~ NA_character_)), 
         rlang::expr(gtsummary::as_flextable()))
  
  # creating then executing code to add footnotes to each p-value
  map2(
    prep_to_create_code$stat_test_lbl, 
    prep_to_create_code$data, 
    function(x, y) {
      rlang::expr(
        flextable::footnote(
          # x = !!y,
          value = as_paragraph(c(!!x)),
          j = "p.value",
          i = ~ label %in% !!y$var_label
          part = "body",
          inline = T
          
        )
      )
    }
  ) %>%
    # concatenating expressions with %>% between each of them
    {c(prepping_calls, .)} %>%    
    purrr::reduce(function(x, y) expr(!!x %>% !!y)) %>%
    # evaluating expressions
    eval()
}


tbl_cases = rota_geno_pheno_cc %>% 
  filter(cc_status == "All rota cases") %>% 
  dplyr::select(age_years, sex, ethnicity, phenotypic_bldgrp_6bg_4bg, ptype_exc_cln, ptype_mx_nt, hospitalization_days, weight, muac, diarrhoea, diarrhoea_durn, vomiting, vomiting_durn, shock, fever, malaria, secretor_status) %>% 
  tbl_summary(., by = secretor_status, missing = "ifany", type = list(all_dichotomous() ~ "categorical")) %>% 
  add_p(pvalue_fun = function(x) style_pvalue(x, digits = 2)) %>% 
  add_stat_label()%>%
  bold_p() #%>% 
  # add_p_footnotes()
  
tbl_ctrls = rota_geno_pheno_cc %>% 
  filter(cc_status == "Controls") %>% 
  dplyr::select(age_years, sex, ethnicity, phenotypic_bldgrp_6bg_4bg, secretor_status) %>%
  tbl_summary(., by = secretor_status, missing = "ifany") %>% 
  add_p(pvalue_fun = function(x) style_pvalue(x, digits = 2)) %>% 
  add_stat_label()%>%
  bold_p()

tbl_demog = tbl_merge(list(tbl_cases, tbl_ctrls),tab_spanner = c("Cases", "Controls")) %>%  
  gtsummary::as_flextable() %>% 
  padding(padding=0) %>% 
  set_table_properties(layout = "autofit") %>% 
  flextable::fontsize(size = 8, part = "all") %>% 
  flextable::border(i = ~ label %in% c("Age (years), median (IQR)", "Sex, n (%)", "Hospitalization duration (Days), median (IQR)", "Weight (Kg), median (IQR)", "Diarrhoea, n (%)", "Diarrhoea duration (Days), median (IQR)", "Vomiting, n (%)", "Vomiting duration, median (IQR)", "Mid upper arm circumference (cm), median (IQR)", "Ethnicity, n (%)", "Exclusive RV P genotype, n (%)", "O VS Non-O, n (%)", "ABO 4, n (%)", "RV P genotype including mixed, n (%)", "Shock, n (%)", "Fever (>37.5C), n (%)", "Malaria co-infection, n (%)", "Mixed & non-typable P genotypes, n (%)"), border.top = fp_border(color="gray70", width = 1), part = "body") %>%
  bold(i = ~ label %in% c("Age (years), median (IQR)", "Sex, n (%)", "Hospitalization duration (Days), median (IQR)", "Weight (Kg), median (IQR)", "Diarrhoea, n (%)", "Diarrhoea duration (Days), median (IQR)", "Vomiting, n (%)", "Vomiting duration, median (IQR)", "Mid upper arm circumference (cm), median (IQR)", "Ethnicity, n (%)", "Exclusive RV P genotypes, n (%)", "O VS Non-O, n (%)", "ABO 4, n (%)", "RV P genotype including mixed, n (%)", "Shock, n (%)", "Fever (>37.5C), n (%)", "Co-infections, n (%)", "Mixed & non-typable P genotypes, n (%)"), j=1, bold = TRUE, part = "body") %>%
  bold(bold = TRUE, part = "header")
  
# a = rota_geno_pheno_cc %>% 
#   filter(cc_status == "All rota cases") %>% 
#   dplyr::select(age_years, sex, hospitalization_days, weight, diarrhoea, diarrhoea_durn, vomiting, vomiting_durn, muac, ethnicity, ptype_exc_cln, non_OvsO_jcr, phenotypic_bldgrp_6bg_4bg, pgenotype, shock, fever, malaria, secretor_status) %>% 
#   tbl_summary(., by = secretor_status, missing = "ifany", type = list(all_dichotomous() ~ "categorical")) %>% 
#   add_p(pvalue_fun = function(x) style_pvalue(x, digits = 2)) %>% 
#   add_stat_label()%>%
#   tbl_stack(list(.,rota_geno_pheno_cc %>% 
#               filter(cc_status == "Controls") %>% 
#               dplyr::select(age_years, sex, non_OvsO_jcr, phenotypic_bldgrp_6bg_4bg, secretor_status) %>%
#               tbl_summary(., by = secretor_status, missing = "ifany") %>% 
#               add_p(pvalue_fun = function(x) style_pvalue(x, digits = 2)) %>% 
#               add_stat_label() ))
# # add_p_footnotes() %>% 
#   gtsummary::as_flextable() %>% 
#   padding(padding=0) %>% 
#   set_table_properties(layout = "autofit") %>% 
#   flextable::fontsize(size = 8, part = "all") 

read_docx("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx") %>% 
  body_add_par(value = "Table 3: Case control participants demographics", style = "heading 2") %>% 
  body_add_flextable(tbl_demog) %>%  
  body_add_par(value = "Tabulation and comparison of cases and controls baseline characteristics and clinical information by secretor status", style = "Table Caption") %>% 
  body_end_section_continuous() %>% 
  print("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx")

# b = rota_geno_pheno_cc %>% 
#   filter(cc_status == "Controls") %>% 
#   dplyr::select(age_years, sex, non_OvsO_jcr, phenotypic_bldgrp_6bg_4bg, secretor_status) %>%
#   tbl_summary(., by = secretor_status, missing = "ifany") %>% 
#   add_p(pvalue_fun = function(x) style_pvalue(x, digits = 2)) %>% 
#   add_stat_label() %>% 
#   gtsummary::as_flextable() %>% 
#   padding(padding=0) %>% 
#   set_table_properties(layout = "autofit") %>% 
#   flextable::fontsize(size = 8, part = "all") %>% 
#   border(i = ~ label %in% c("Age (years), median (IQR)", "Sex, n (%)", "Hospitalization duration (Days), median (IQR)", "Weight (Kg), median (IQR)", "Diarrhoea, n (%)", "Diarrhoea duration (Days), median (IQR)", "Vomiting, n (%)", "Vomiting duration, median (IQR)", "Mid upper arm circumference (cm), median (IQR)", "Ethnicity, n (%)", "Exclusive RV P genotype, n (%)", "O VS Non-O, n (%)", "ABO 4, n (%)", "RV P genotype including mixed, n (%)", "Shock, n (%)", "Fever (>37.5C), n (%)", "Malaria co-infection, n (%)"), border.top = fp_border(color="gray70", width = 1), part = "body") %>%
#   bold(i = ~ label %in% c("Age (years), median (IQR)", "Sex, n (%)", "Hospitalization duration (Days), median (IQR)", "Weight (Kg), median (IQR)", "Diarrhoea, n (%)", "Diarrhoea duration (Days), median (IQR)", "Vomiting, n (%)", "Vomiting duration, median (IQR)", "Mid upper arm circumference (cm), median (IQR)", "Ethnicity, n (%)", "Exclusive RV P genotype, n (%)", "O VS Non-O, n (%)", "ABO 4, n (%)", "RV P genotype including mixed, n (%)", "Shock, n (%)", "Fever (>37.5C), n (%)", "Malaria co-infection, n (%)"), j=1, bold = TRUE, part = "body") %>%
#   bold(bold = TRUE, part = "header")
#   
# 
# read_docx("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx") %>% 
#   body_add_par(value = "Table 1", style = "heading 2") %>% 
#   body_add_flextable(a) %>%  
#   body_add_par(value = "Table: ", style = "Table Caption") %>% 
#   body_end_section_continuous() %>% 
#   print("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx")


# summary(tableby(interaction(secretor_status, surveillance_period) ~ hospitalization_days + weight + diarrhoea + diarrhoea_durn + vomiting + vomiting_durn + muac + ethnicity + ptype_exc_cln + non_OvsO_jcr + phenotypic_bldgrp_6bg_4bg + pgenotype + shock + fever + malaria, data = rota_geno_pheno_cc[rota_geno_pheno_cc[,"cc_status"] == "All rota cases",], numeric.stats = c("Nmiss", "median", "iqr", "range", "medianmad", "medianq1q3"), cat.stats=c("Nmiss", "countrowpct")), pfootnote = T, digits = 2, digits.pct =2, conf.level = 0.95)
# summary(tableby(surveillance_year ~ ptype_exc_cln * secretor_status,data = rota_geno_pheno_cc, numeric.stats = c("Nmiss", "median", "iqr", "range", "medianmad", "medianq1q3"), cat.stats=c("Nmiss", "countrowpct")), pfootnote = T, digits = 2, digits.pct =2, conf.level = 0.95)

# summary(tableby(ptype_exc_cln ~ age_years + hospitalization_days + weight + diarrhoea + vomiting + muac + ethnicity + secretor_status + non_OvsO_jcr + phenotypic_bldgrp_6bg_4bg + pgenotype + lewis_bg + lewis_pos ,data = rota_geno_pheno_cc, numeric.stats = c("Nmiss", "median", "iqr", "range", "medianmad", "medianq1q3"), cat.stats=c("Nmiss", "countrowpct")), pfootnote = T, digits = 2, digits.pct =2, conf.level = 0.95)
```

##Effect of secretor status and ABO blood groups without factoring in surveillance period or years {.tabset}

```{r echo=F, results= 'asis', warning=F, message=F, eval=F}
panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)

ps <- prop_section(page_size = page_size(orient = "landscape"),page_margins = page_mar(left = 0.2, right = 0.2),type = "nextPage")

#within everyone
snp_log_rgrsn_everyone = function (predctr, predctr_q) {
  model_list=list()
  count_list=list()
  mal_list=list()
  
  for (i in 1:length(rota_strain[c(1,2)])){
    all_tbl = rota_geno_pheno_cc %>% filter(malaria != "Malaria" | is.na(malaria)) %>% dplyr::select(!!predctr_q,!!rota_strain_q[[i]], "ethnicity")
    
    ##rota strain ~ abo/fut2 + ethnicity in all plus those with malaria
    model_list[[length(model_list)+1]] = all_tbl%>% dplyr::select(-ethnicity) %>% tbl_summary(., by = rota_strain[c(1,2)][i]) %>% list(., tbl_regression(glm(as.formula(paste(rota_strain[c(1,2)][i], "~",  predctr )), data = all_tbl, family = "binomial"), pvalue_fun = function(x) style_pvalue(x, digits = 2), exponentiate = T)) %>% tbl_merge()
    
    ##Adjusted
    model_list[[length(model_list)+1]] = all_tbl%>% filter(!is.na(ethnicity)) %>% dplyr::select(-ethnicity) %>% tbl_summary(., by = rota_strain[c(1,2)][i]) %>% list(., tbl_regression(glm(as.formula(paste(rota_strain[c(1,2)][i], "~",  predctr, "+" ,  paste("ethnicity", collapse = "+") )), data = all_tbl, family = "binomial"), include = -ethnicity, pvalue_fun = function(x) style_pvalue(x, digits = 2), exponentiate = T)) %>% tbl_merge()
    
  }
  
  cat("\n")    
  
  
  
  #}
  #}
  
  tryCatch({
    cat(paste("<B><U> All rotavirus strains ~ ",  paste(predctr, collapse = " + "), "+" , paste("ethnicity", collapse = " + "), "</U></B> \n \n"))
    
    a = gtsummary::as_flextable(tbl_merge(model_list, tab_spanner = paste(rep(c("Unadjusted", "Adjusted (ethnicity)"), times = length(rota_strain[c(1,2)])),rep(rota_strain[c(1,2)], times = 1, each = 2))), size = 8) %>%  flextable::fontsize(size = 8, part = "all") %>%
      padding(padding=0) %>%
      set_table_properties(layout = "autofit")
    
    read_docx("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx") %>% body_add_par(value = predctr, style = "heading 2") %>% body_add_flextable(a) %>%  body_add_par(value = "Table 1: ", style = "Table Caption") %>% body_end_block_section(block_section(ps)) %>% print("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx")   
    a
  }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  
  
  
}

# cat("\n \n")
# cat("###4 ABO groups\n")
# snp_log_rgrsn_everyone(predctr = "phenotypic_bldgrp_6bg_4bg", predctr_q = quo(phenotypic_bldgrp_6bg_4bg))

cat("\n \n")
cat("###secretor status \n") 
snp_log_rgrsn_everyone(predctr = "secretor_status", predctr_q = quo(secretor_status))
cat("\n \n")

```

##Effect of secretor status and ABO blood groups without factoring in surveillance period or years among secretors {.tabset}

```{r echo=F, results= 'asis', warning=F, message=F}
panderOptions('knitr.auto.asis', FALSE) 
panderOptions('table.split.table', Inf)


#within everyone
snp_log_rgrsn_secretor = function (predctr, predctr_q) {
  model_list=list()
  count_list=list()
  mal_list=list()
  
  for (i in 1:length(rota_strain[c(1,2)])){
    sec_tbl = rota_geno_pheno_cc %>% filter(secretor_status == "secretor" & (malaria != "Malaria" | is.na(malaria))) %>% dplyr::select(!!predctr_q,!!rota_strain_q[c(1,2)][[i]], "ethnicity")
    
    ##rota strain ~ abo/fut2 + ethnicity in all plus those with malaria
    model_list[[length(model_list)+1]] = sec_tbl %>% dplyr::select(-ethnicity) %>% tbl_summary(., by = rota_strain[c(1,2)][i]) %>% list(., tbl_regression(glm(as.formula(paste(rota_strain[c(1,2)][i], "~",  predctr )), data = sec_tbl, family = "binomial"), pvalue_fun = function(x) style_pvalue(x, digits = 2), exponentiate = T) %>%  bold_p()) %>% tbl_merge() 
    
    ##Adjusted
    model_list[[length(model_list)+1]] = sec_tbl%>% filter(!is.na(ethnicity)) %>% dplyr::select(-ethnicity) %>% tbl_summary(., by = rota_strain[c(1,2)][i]) %>% list(., tbl_regression(glm(as.formula(paste(rota_strain[c(1,2)][i], "~",  predctr, "+" ,  paste("ethnicity", collapse = "+") )), data = sec_tbl, family = "binomial"), include = -ethnicity, pvalue_fun = function(x) style_pvalue(x, digits = 2), exponentiate = T) %>%  bold_p()) %>% tbl_merge()
    
  }
  
  cat("\n")    
  
  
  
  #}
  #}
  
  tryCatch({
    cat(paste("<B><U> All rotavirus strains ~ ",  paste(predctr, collapse = " + "), "+" , paste("ethnicity", collapse = " + "), "</U></B> \n \n"))
    
    a = gtsummary::as_flextable(tbl_merge(model_list, tab_spanner = paste(rep(c("Unadjusted", "Adjusted (ethnicity)"), times = length(rota_strain[c(1,2)])),rep(rota_strain[c(1,2)], times = 1, each = 2))), size = 8) %>%  
      flextable::fontsize(size = 8, part = "all") %>%
      padding(padding=0) %>%
      set_table_properties(layout = "autofit")
    
    read_docx("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx") %>% body_add_par(value = predctr, style = "heading 2") %>% body_add_flextable(a) %>%  body_add_par(value = "Table 1: ", style = "Table Caption") %>% body_end_block_section(block_section(ps)) %>% print("C:/Users/Jesse Rop/OneDrive - Kemri Wellcome Trust/WT 18 month project/JRWT Study data/rotavirus analysis/rotavirus/rotavirus_git/rota_figures_and_tables.docx")
    
    a
  }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  
  
  
}

cat("\n \n")
cat("###4 ABO groups\n")
snp_log_rgrsn_secretor(predctr = "phenotypic_bldgrp_6bg_4bg", predctr_q = quo(phenotypic_bldgrp_6bg_4bg))
cat("\n \n")

cat("\n \n")
cat("###6 ABO groups\n")
snp_log_rgrsn_secretor(predctr = "phenotypic_bldgrp_6bg", predctr_q = quo(phenotypic_bldgrp_6bg))
cat("\n \n")

```

